feat: Introduce compose logs subcommand

LucaSeri opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes
Introduce the ``kraft compose logs`` subcommand that allows users to print existing logs or follow them.

E.g:

```bash
kraft compose up -d

kraft compose logs --follow

ping-pong-pong  | Received: ping
ping-pong-ping  | Received: pong
ping-pong-pong  | Received: ping
```

<!--
Please provide a detailed description of the changes made in this new PR.
-->

craciunoiuc commented on internal/cli/kraft/compose/logs/logs.go:
```diff
@@ -0,0 +1,112 @@
+	return opts.PrintLogs(ctx, project)
+}
+
+func (opts *LogsOptions) PrintLogs(ctx context.Context, project *compose.Project) error {
+	var wg sync.WaitGroup
```
shouldn't this be private? Is it used anywhere else?

`printLogs`

craciunoiuc replied:
Nvm, saw it used  in some other place also below.

Move it please to a `internal/cli/kraft/compose/utils/utils.go` file and make public there

craciunoiuc commented on internal/cli/kraft/compose/logs/logs.go:
```diff
@@ -0,0 +1,112 @@
+			longestName = len(service.Name)
+		}
+	}
+	for i := range project.Services {
```
wouldn't it look better to just do:

`for _, service := range project.Services {`

and then

`(service)`

?

craciunoiuc replied:
Actually even better, only the name seems to be used, so

`(service.Name)` and then use a `name` variable inside everywhere

craciunoiuc reviewed (changes requested):
left you some comments inline, overall looks fine

Commit:
feat(compose): Add logs subcommand

The logs subcommand allows printing and following
the logs generated by a compose project.

Signed-off-by: Luca Seritan <luca.seritan@gmail.com>

LucaSeri commented:
Thanks @craciunoiuc for the comments. Part of this PR was actually migrated to https://github.com/unikraft/kraftkit/pull/1402, and ``kraft compose logs`` is now almost trivial.

nderjung reviewed (approved):
Thanks!

craciunoiuc reviewed (approved):
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
